### ISSUE TITLE
[bug] no-missing-key false positive

### ISSUE THREAD

**Issue by leetcow** [Rel1cx/eslint-react#986]
### Describe the bug

<img width="895" alt="Image" src="https://github.com/user-attachments/assets/c3b73ab6-86d2-40d7-8791-1966530c8828" />

You can see the error being triggered here even though the "keys" variable is not used anywhere in the component.

### Platform and versions

```Text
'@eslint-react/eslint-plugin': '^1.32.0',
```

### Stack trace

```Text

```
